Polyhedron/include/CGAL/boost/graph/properties_Polyhedron_3.h

```diff
@@ -30,13 +30,63 @@
 #include <CGAL/number_utils.h>
 #include <boost/shared_ptr.hpp>
 #include <CGAL/boost/graph/internal/Has_member_id.h>
+#include <boost/unordered_map.hpp>
 
 #define CGAL_HDS_PARAM_ template < class Traits, class Items, class Alloc> class HDS
 
 namespace CGAL {
 
 namespace internal {
 
+template <typename Poly, typename K, typename V>
+struct Dynamic_polyhedron_property_map {
+
+  typedef K key_type;
+  typedef V value_type;
+  typedef value_type& reference;
+  typedef boost::read_write_property_map_tag  category;
+
+ 
+
+  Dynamic_polyhedron_property_map(const V& default_value)
+    : map_(new Map()), default_value(default_value)
+  {}
+
+  void clear()
+  {
+    map_ = boost::shared_ptr<Map>(0);
+  }
+
+
+  friend value_type get(const Dynamic_polyhedron_property_map& m, const key_type& k)
+  {
+    typename Map::const_iterator it = m.map_->find(k);
+    if(it == m.map_->end()){
```

Comment:
The lazy behavior that uses `default()` in `get` and `put` seems correct to me, but I am worried we may find a configuration where it's not correct anymore.
Maybe we could make this behavior optional, with a `Lazy_tag` template parameter?